app/controllers/posts_controller.rb

```diff
@@ -4,9 +4,28 @@ def index
     @posts = @user.posts unless @user.nil?
   end
 
+  def new
+    @post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
   def show
     @user = User.find_by_id(params[:user_id])
     @post = Post.find_by_id(params[:id]) unless @user.nil?
     @comments = @post.comments unless @post.nil?
   end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
```

Comment:
- [ ] [optional] It would be nice if you have strong parameters for your post controller, this requires having a private method for parameters and calling the method when in need to use the params, this will make your app more safer.